### ISSUE TITLE
Hardware decoding not working on Sony Android TV

### ISSUE THREAD

**Issue by danaigus** [mpv-android/mpv-android#300]
Tested HEVC 8/10bit and AVC 8bit, mpv only uses software decoding in all cases.
[logcat](https://gist.github.com/danaigus/68188d551aa45b27e4f212ce36559e16)

**Comment by sfan5** [comment]
10bit AVC not being support isn't surprising, but 10bit HEVC not working might be an issue on ffmpeg's side.
Logcat shows `hevc_mediacodec: Output color format 0x7f000103 (value=2130706691) is not supported` but this format is nowhere to be found in [Google's documentation](https://developer.android.com/reference/android/media/MediaCodecInfo.CodecCapabilities#constants_1).
Documentation for [MediaCodec.getOutputFormat](https://developer.android.com/reference/android/media/MediaCodec#getOutputFormat()) suggests this is just some default format that can be ignored because it changes once decoding actually starts (can it?).

Do you have a log for AVC 8bit not working?

**Comment by danaigus** [comment]
Here is the [logcat](https://gist.github.com/danaigus/5a496b58c752c565b8f344bad5f243f8) for AVC 8bit.

**Comment by sfan5** [comment]
Looks like the same issue as with HEVC.
Here's an APK with a possible fix based on my guess earlier: https://kitsunemimi.pw/tmp/mpv-android-2020-08-31.apk

**Comment by danaigus** [comment]
It's still using software decoding. Here's the [logcat](https://gist.github.com/danaigus/9c51ada1523eeff60372e270c9576283) from the same AVC 8bit file.

**Comment by harry71019** [comment]
I also tried and I find that it doesn't HW decode a 4K hevc 10bit 60 fps stream.

**Comment by blahblah60** [comment]
I got Sony X75H mpv will always use SW
I tried playing x265 10/8 bit encode and won't play properly. **Green distorted images**.
I tried playing x264 8bit but using SW and its laggy.
I tried playing x264 10bit and won't play.

**Comment by sfan5** [comment]
Here's another APK you all can test: https://kitsunemimi.pw/tmp/mpv-android-2021-02-11.apk

Please report back whether hardware decoding works and provide a logcat in either case.

**Comment by sfan5** [comment]
Yes a log would be immensely helpful.
You'll need to `adb` on your computer, connect your phone and [this page](https://developer.android.com/studio/command-line/logcat#Syntax) may be helpful on running logcat.

**Comment by danaigus** [comment]
Tested two low bit rate files (around 2 Mb/s) , one 720p AVC 8bit and the other 1080p HEVC 10bit, and and they seem to run smoothly. But when I tried playing a high bit rate AVC 8bit (29 Mb/s) many frames were dropped, and with a high bit rate HEVC 10bit (60 Mb/s) the video was freezing most of the time.
Here are the logcat for the two files: [hevc10bit-4k-hdr](https://gist.github.com/danaigus/f475250b00061a74f00724bbd257af08) - [avc8bit](https://gist.github.com/danaigus/d0a82f1d4d67f455b1bf770e3dfa6552)

**Comment by sfan5** [comment]
Good to know that hwdec works with that patch in principle, the high bitrate problem likely exists independent of this issue.

**Comment by Icemic** [comment]
Hi, I've tried 2021-02-11 version on my Sony X75 90J and get a similar result to danaigus's. It's always showing a black screen with AVC 8bit or a quick flashing between freezing frame and black.
Besides, It often crashes after some pause-resume operations.

What's interesting, if I saw a black screen, then just to pause and press 'right' to perform a fast forward, a still frame can be seen at the time. So I guess this issue may be related to OSD rendering?

If there is any progress here, I am willing to help.
